hazelcast/src/main/java/com/hazelcast/spi/impl/BinaryOperationFactory.java

```diff
@@ -28,6 +28,10 @@
 
 import java.io.IOException;
 
+/**
+ * An {@link OperationFactory} that creates operations by cloning
```

Comment:
Shouldn't we rename it to `CloningOperationFactory` then? Now it sounds like it's creating `BinaryOperations`, which doesn't make sense.

Also the `NodeEngine` field should be `transient`.

Reply:
Good comments, but that should be part of a different PR. This PR is ultra low risk since it is just adding javadoc.